self_testim.py

```diff
@@ -0,0 +1,41 @@
+def is_valid(board, neighbor, initial_point):  # neighbor[0]=row index, neighbor[1]=col index,
+    return 0 <= neighbor[0] < len(board) and 0 <= neighbor[1] < len(board[neighbor[0]]) \
+           and board[neighbor[0]][neighbor[1]] != -1 and [neighbor[0], neighbor[1]] > initial_point
+
+
+def solve_maze_monotonic_rec(board, initial_point, goal_point, solution_path):
+    if initial_point == goal_point:  # solution found
+        solution_path.append(initial_point)
+        return solution_path
+    row = initial_point[0]
+    col = initial_point[1]
+    if is_valid(board, (row + 1, col), initial_point):
+        solution_path.append(initial_point)
+        solve_maze_monotonic_rec(board, [row + 1, col], goal_point, solution_path)
+    elif is_valid(board, (row, col + 1), initial_point):
+        solution_path.append(initial_point)
```

Comment:
You are using the same object `solution_path` for each one of the `if`s, they will ruin each other. You should copy the `solution_path` or delete point if solution wasn't found.